dora_node_hub/dora_gemini_diff_drive_navigation/dora_gemini_diff_drive_navigation/main.py

```diff
@@ -0,0 +1,138 @@
+    # Optional node configuration
+    model = os.getenv("MODEL", "gemini-2.5-flash")
+
+    # Initialize the controller with the specified model.
+    controller = DiffDriveGeminiControl(model=model)
+    last_image = None
+    # Initialize cmd_vel with zero velocity.
+    cmd_vel = zero_twist()
+
+    while True:
+        event = await node.recv_async()
+        event_type = event["type"]
+        if event_type == "INPUT":
+            event_id = event["id"]
+            if event_id == "image":
+                storage = event["value"]
+                metadata = event["metadata"]
+                last_image = image_data_to_png(storage, metadata)
+            if event_id == "tick":
+                if last_image is None:
+                    continue
+                if command == "":
+                    cmd_vel = zero_twist()
+                    continue
+                image_bytes = last_image
+
+                # For debugging: Dump image into a file for debugging
+                # with open("last_image.png", "wb") as f:
+                #    f.write(last_image)
+
+                generated_velocities_result = await controller.generate_velocities(
+                    command=command, image_bytes=image_bytes
+                )
+                print(f"Generated velocities: {generated_velocities_result}")
+                if command.lower() == "stop" and (np.all(generated_velocities_result == zero_twist())):
+                    # To avoid keeping generating velocities in the next iteration when the command is "stop",
+                    # we update the command to an empty string.
+                    command = ""
+                last_image = None
+                cmd_vel = generated_velocities_result
+            if event_id == "cmd_vel_tick":
+                node.send_output(
+                    "cmd_vel",
+                    data=pa.array(cmd_vel.tolist(), type=pa.float64()),
+                    metadata=event["metadata"],
+                )
+            if event_id == "command":
+                value = event["value"]
+                # value is a pa.array of type pa.string()
+                if not isinstance(value, pa.Array):
+                    raise RuntimeError(f"Expected value to be a pyarrow Array, got {type(value)}")
+                if value.type != pa.string():
+                    raise RuntimeError(f"Expected value to be a pyarrow string Array, got {value.type}")
+                value = value.to_pylist()
+                if len(value) > 1:
+                    raise RuntimeError(f"Expected value to be a single string, got {value.len()} strings")
+                command = str(value[0])
+
+                print(f"Received command: {command}")
+                if command == "":
+                    print("Command is empty. Waiting for next input event.")
+                    continue
+
+
+loop = asyncio.get_event_loop()
+loop.run_until_complete(main())
```

Comment:
@francocipollone uh, no `__main__` check?: 
```suggestion
if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    loop.run_until_complete(main())
```